src/system.rs

```diff
@@ -181,13 +181,41 @@ fn interpret_turbo(is_turbo: &mut String) -> Result<bool, Error> {
     }
 }
 
+fn read_bat_energy_full(design: bool) -> Result<i32, Error> {
+    let mut capacity_readings: String = String::new();
+    let bat_energy_full_path: &str = "/sys/class/power_supply/BAT0/";
+    if design == true {
+        File::open(bat_energy_full_path.to_string() + "energy_full_design")?
+            .read_to_string(&mut capacity_readings)?;
+    } else {
+        File::open(bat_energy_full_path.to_string() + "energy_full")?
+            .read_to_string(&mut capacity_readings)?;
+    }
+    Ok(capacity_readings.parse::<i32>().unwrap())
```

Comment:
hmm im getting a `thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }'` here

Reply:
capacity_readings might have a newline and then cannot be interpreted as a int

Reply:
you can just capacity_readings.pop() it

Reply:
oooo okay thanks